### PR TITLE
Features/109 add events backend

### PR DESCRIPTION
**Pull request by bomanviktor**
Pull request for #109:

Endpoints added:

For groups:
- GET `/api/groups/(\d+)/events` -> event ids of a group
- POST `/api/groups/(\d+)/events/create` -> create the event, returns event id
 
New event endpoints:
- GET `/api/events/id` -> response body of event
- GET `/api/events/id/members` -> members that have "accepted" the event
- POST `/api/events/id/leave` -> leave the event, returns nil
- POST `/api/events//id/going` -> "accept" the event, used if leave has been used

Due to my previous function to check if user already exists in the event did not work (return true even if user does not exist in event), I added a `UNIQUE` constraint to the `event_members` table. This will result in the user getting `200 OK` even if user exists/does not exist in the table.

### CODE REVIEW

**Comment by maximihajlov** [comment]
We need a `status` field in `/api/events/{id}` similar to `memberStatus`. 0 if not set, 1 if coming, 2 if not coming

File: backend/forum/database/ops_events.go [hunk 1]
```diff
@@ -128,15 +129,14 @@ func (db DB) GetEventStatus(eventId, userId int) InviteStatus {
 	)
 	END 
 	AS event_status
-	`, eventId, userId)
+	`, sql.Named("eventId", eventId), sql.Named("userId", userId))
```
Comment (maximihajlov) [inline backend/forum/database/ops_events.go:132]: @bomanviktor, @FinnTune. That's why it's important to use `sql.Named`. Here you placed `eventId` first, which caused inserting `eventId` into the first parameter, which is `:userId` and not  `:eventId`. So it worked only if `userId` and `eventId` are the same

Comment (maximihajlov) [reply backend/forum/database/ops_events.go:132]: To be able to see all the possible options:
image

It's a common practice in Go:

```go
const (
	MethodGet     = "GET"
	MethodHead    = "HEAD"
	MethodPost    = "POST"
	MethodPut     = "PUT"
	MethodPatch   = "PATCH" 
	MethodDelete  = "DELETE"
	MethodConnect = "CONNECT"
	MethodOptions = "OPTIONS"
	MethodTrace   = "TRACE"
)

// and not Get = "Get"
```

Comment (maximihajlov) [reply backend/forum/database/ops_events.go:132]: In other languages with better implementation of enums they look like this:

```Java
enum Level {
  LOW,
  MEDIUM,
  HIGH
}

Level myVar = Level.MEDIUM;
```

But in go for some reason possible constants are not associated with the type, so using prefix is the only way to get this dev experience.

Comment (maximihajlov) [reply backend/forum/database/ops_events.go:132]: https://go.dev/issue/19814

**Comment by maximihajlov** [comment]
Looks good for me. It's stable enough to be merged with frontend.

**Comment by bomanviktor** [comment]
> Looks good for me. It's stable enough to be merged with frontend.

Awesome! LFG!

**Pull request merged**